CI Failures: jarHell failed to run on Windows

This jarHell failure has cropped up a few times in the last couple days on master and 7.4 branches. 

https://groups.google.com/a/elastic.co/forum/#!searchin/build-elasticsearch/jarHell%7Csort:date

All with the same failure: 

```
14:41:02 > Task :x-pack:plugin:ml:jarHell FAILED
14:41:02 Exec output and error:
14:41:02 | Output for C:\Users\jenkins\.java\openjdk12\bin\java.exe:
14:41:02 
14:41:02 > Task :x-pack:plugin:security:testingConventions
14:41:02 
14:41:02 FAILURE: Build failed with an exception.
14:41:02 
14:41:02 * What went wrong:
14:41:02 Execution failed for task ':x-pack:plugin:ml:jarHell'.
14:41:02 > A problem occurred starting process 'command 'C:\Users\jenkins\.java\openjdk12\bin\java.exe
```

Here's a scan: https://gradle-enterprise.elastic.co/s/4dnt4bscjqldo 
The root  cause: 
```
Caused by: java.io.IOException: CreateProcess error=206, The filename or extension is too long
```

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/precommit/JarHellTask.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/precommit/JarHellTask.java
@@ -42,7 +42,7 @@ public class JarHellTask extends PrecommitTask {
     @TaskAction
     public void runJarHellCheck() {
         LoggedExec.javaexec(getProject(), spec -> {
-            spec.classpath(getClasspath());
+            spec.environment("CLASSPATH", getClasspath().getAsPath());
             spec.setMain("org.elasticsearch.bootstrap.JarHell");
         });
     }

